packages/wagmi/src/client.ts

```diff
@@ -430,11 +455,16 @@ export class Web3Modal extends Web3ModalScaffold {
       await Promise.all([
         this.syncProfile(address, chainId),
         this.syncBalance(address, chainId),
-        this.syncConnectedWalletInfo(connector),
         this.setApprovedCaipNetworksData()
       ])
+      if (connector) {
+        this.syncConnectedWalletInfo(connector)
+      }
+      if (addresses?.length) {
+        this.setAllAccounts(addresses.map(addr => ({ address: addr, type: 'eoa' })))
```

Comment:
Do we sure this wouldn't break the SA flows by forcing the accounts switch to EOA? Like if there is any changes to accounts, we are updating them with EOA type, but what about the SA ones?

Reply:
Good point, will take a look as I just moved code around.